app/lib/features/chat_ng/widgets/reply_preview.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:acter/features/chat_ng/models/reply_message_state.dart';
+import 'package:acter/features/chat_ng/providers/chat_room_messages_provider.dart';
+import 'package:acter/features/chat_ng/widgets/events/reply_original_event.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:skeletonizer/skeletonizer.dart';
+
+// Reply State UI widget
+class RepliedToPreview extends ConsumerWidget {
+  final String roomId;
+  final String originalId;
+  final bool isUser;
+  const RepliedToPreview({
+    super.key,
+    required this.roomId,
+    required this.originalId,
+    this.isUser = false,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final RoomMsgId replyInfo = (roomId: roomId, uniqueId: originalId);
+
+    final roomMsgState = ref.watch(replyToMsgProvider(replyInfo)).valueOrNull;
+    final avatarInfo = AvatarInfo(uniqueId: '#');
+
+    return switch (roomMsgState) {
+      ReplyMsgLoading() => replyBuilder(
+          context,
+          Skeletonizer(
+            child: ListTile(
+              leading: ActerAvatar(options: AvatarOptions.DM(avatarInfo)),
```

Comment:
only needed for the loading case, no reason to generate it otherwise.
```suggestion
    return switch (roomMsgState) {
      ReplyMsgLoading() => replyBuilder(
          context,
          Skeletonizer(
            child: ListTile(
              leading: ActerAvatar(options: AvatarOptions.DM(AvatarInfo(uniqueId: '#'))),
```